custom_components/tuya_local/devices/probreeze_pb20.yaml

```diff
@@ -0,0 +1,165 @@
+      type: integer
+      range:
+        min: 30
+        max: 80
+      mapping:
+        - step: 5
+    - id: 11
+      name: error
+      type: bitfield
+      mapping:
+        - dps_val: 0
+          value: OK
+        - dps_val: 8
+          value: tank
+secondary_entities:
+  - entity: fan
+    dps:
+      - id: 1
+        type: boolean
+        name: switch
+      - id: 6
+        name: speed
+        type: string
+        mapping:
+          - dps_val: "0"
+            value: 100
+          - dps_val: "1"
+            value: 50
+  - entity: lock
+    name: Child lock
+    category: config
+    dps:
+      - id: 7
+        type: boolean
+        name: lock
+        mapping:
+          - dps_val: false
+            icon: "mdi:hand-back-right"
+          - dps_val: true
+            icon: "mdi:hand-back-right-off"
+  - entity: select
+    name: Countdown
+    icon: "mdi:timer"
+    category: config
+    dps:
+      - id: 12
+        name: option
+        type: string
+        mapping:
+          - dps_val: "0"
+            value: Cancel
+          - dps_val: "1"
+            value: "1 hour"
+          - dps_val: "2"
+            value: "2 hours"
+          - dps_val: "3"
+            value: "3 hours"
+          - dps_val: "4"
+            value: "4 hours"
+          - dps_val: "5"
+            value: "5 hours"
+          - dps_val: "6"
+            value: "6 hours"
+          - dps_val: "7"
+            value: "7 hours"
+          - dps_val: "8"
+            value: "8 hours"
+          - dps_val: "9"
+            value: "9 hours"
+          - dps_val: "10"
+            value: "10 hours"
+          - dps_val: "11"
+            value: "11 hours"
+          - dps_val: "12"
+            value: "12 hours"
+          - dps_val: "13"
+            value: "13 hours"
+          - dps_val: "14"
+            value: "14 hours"
+          - dps_val: "15"
+            value: "15 hours"
+          - dps_val: "16"
+            value: "16 hours"
+          - dps_val: "17"
+            value: "17 hours"
+          - dps_val: "18"
+            value: "18 hours"
+          - dps_val: "19"
+            value: "19 hours"
+          - dps_val: "20"
+            value: "20 hours"
+          - dps_val: "21"
+            value: "21 hours"
+          - dps_val: "22"
+            value: "22 hours"
+          - dps_val: "23"
+            value: "23 hours"
+          - dps_val: "24"
+            value: "24 hours"
+  - entity: sensor
+    name: Remaining Time
+    class: duration
+    dps:
+      - id: 13
+        type: integer
+        name: sensor
+        unit: min
+  - entity: binary_sensor
+    name: Tank
+    class: problem
+    icon: "mdi:cup-outline"
+    icon-priority: 2
+    category: diagnostic
+    dps:
+      - id: 11
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 8
+            value: true
+            icon: "mdi:cup-water"
+            icon-priority: 1
+          - value: false
+  - entity: switch
```

Comment:
How does this interact with the presets? Does it override and always use continuous?  Maybe it is better to merge this with the presets if it does using `constraint` and `conditions`.

Reply:
Then it makes more sense to merge it with "on - off". So that it will be "off - on - int. drying" Because physical button "internal drying" turns the device on and modes "continuous" and "auto" are not working when drying is active.

Reply:
You can't add additional options to off/on.  You can add presets.

  The following also takes advantage of some translations that Home Assistant provides.

```
    - id: 2
      name: mode
      type: string
      mapping:
        - constraint: inside_drying
          conditions:
            - dps_val: false
              mapping:
                - dps_val: "1"
                  value: boost
                - dps_val: "0"
                  value: auto
            - dps_val: true
              value: laundry
```

Reply:
Ok, understood

Reply:
It should show as a preset.

Reply:
I see only "boost" and "auto":

![image](https://github.com/make-all/tuya-local/assets/5202136/863fa182-ae58-4d27-a31c-4c7aac40a0f1)

Reply:
I need to take a look at how the code handles these multiply nested mappings.  Maybe it will work better to make 101 the mode, and 2 something else like work_mode, to bring things down a level.
```
    - id: 2
      name: work_mode
      type: string
      hidden: true
    - id: 101
      type: boolean
      name: mode
      mapping:
        - dps_val: false
          constraint: work_mode
          conditions:
            - dps_val: "1"
              value: boost
            - dps_val: "0"
              value: auto
        - dps_val: true
          value: laundry
```

Reply:
This works, but "laundry" or "Dry clothes" is not applicable for it. This mode is to make the internal parts of the device dry.

![image](https://github.com/make-all/tuya-local/assets/5202136/a9150e9c-1348-4a16-8bf6-d5d435e184c2)